processing/src/main/java/org/apache/carbondata/processing/loading/steps/CarbonRowDataWriterProcessorStepImpl.java

```diff
@@ -308,7 +312,7 @@ private void processBatch(CarbonRowBatch batch, CarbonFactHandler dataHandler, i
       }
       writeCounter[iteratorIndex] += batch.getSize();
     } catch (Exception e) {
-      throw new CarbonDataLoadingException("unable to generate the mdkey", e);
+      throw new CarbonDataLoadingException(e);
```

Comment:
I think there is no need to wrap the exception here, just remove the try-catch code.

Reply:
The KeyGenException extend Exception, it needs CarbonDataLoadingException(RuntimeException) to wrap and throw.